### ISSUE TITLE
MC/DC LLVM 19 intrinsics changes

### ISSUE THREAD

**Issue by krasimirgg** [rust-lang/rust#126672]
In LLVM 19 some details about MC/DC changed, removing and updating some intrinsics:
* https://github.com/llvm/llvm-project/commit/85a7bba7d28365ff98dae74f20ebf9f53d42023a
In  instrprof_mcdc_condbitmap_update intrinsic was dropped recently:
* https://github.com/llvm/llvm-project/pull/95496

I tried naively to remove the reference to the removed intrinsic https://github.com/rust-lang/rust/pull/126582, but that's incomplete. Will need more work to adapt to this it seems:
* https://github.com/llvm/llvm-project/pull/95496#issuecomment-2173006812

**Comment by krasimirgg** [comment]
@ZhuUx

**Comment by ZhuUx** [comment]
As far as I know, works on mcdc need to be done for adapting to llvm 19 are:

1. Eliminate the removed intrinsic function as you have done.
2. Change the way to update condition bits as [the new mechanism](https://github.com/llvm/llvm-project/pull/82448).
3. Let ConditionId start from 0 rather than 1.

This changes are definite since they have been merged into llvm now, while still there are other changes may be introduced.

Given that `2` actually changes several arguments we passed to llvm, I'd wish to stop supporting mcdc on llvm-18 once if we switched to 19.

**Comment by Zalathar** [comment]
In the short-term, we can just adjust the tests to not run on LLVM 19.

After the in-tree LLVM has been updated to LLVM 19, then we can remove support for MC/DC under LLVM 18, and add support for MC/DC under LLVM 19.

(Unless someone wants to do the extra work of supporting MC/DC with LLVM 19 before the official upgrade.)

**Comment by workingjubilee** [comment]
> After the in-tree LLVM has been updated to LLVM 19, then we can remove support for MC/DC under LLVM 18, and add support for MC/DC under LLVM 19.

This sounds acceptable if we defer stabilizing MC/DC until our minimum LLVM is 19+

**Comment by ZhuUx** [comment]
Filed #126733 to fix it and prepare for llvm 19.

**Comment by nikic** [comment]
If all goes well, I expect we'll try to land the LLVM 19 update next week, with broken MC/DC support if it's not ready in time.

**Comment by ZhuUx** [comment]
The work has been done at #126733 . But it's blocked by #126677 , which has not been reviewed by the assignee yet.
